tfhe/src/shortint/ciphertext/mod.rs

```diff
@@ -85,6 +85,49 @@ pub struct Ciphertext {
     pub pbs_order: PBSOrder,
 }
 
+// #[cfg(feature = "safe_serialization")]
+impl crate::safe_serialization::ParameterSetConformant for Ciphertext {
+    type ParameterSet = super::PBSParameters;
+
+    fn conformant(&self, param: &Self::ParameterSet) -> bool {
+        let (expected_dim, message_modulus, carry_modulus, ciphertext_modulus);
+
+        match param {
+            super::PBSParameters::PBS(param) => {
+                expected_dim = match self.pbs_order {
+                    PBSOrder::KeyswitchBootstrap => {
+                        param.glwe_dimension.0 * param.polynomial_size.0
+                    }
+                    PBSOrder::BootstrapKeyswitch => param.lwe_dimension.0,
+                };
+
+                message_modulus = param.message_modulus;
+                ciphertext_modulus = param.ciphertext_modulus;
+                carry_modulus = param.carry_modulus
+            }
+            super::PBSParameters::MultiBitPBS(param) => {
+                expected_dim = match self.pbs_order {
+                    PBSOrder::KeyswitchBootstrap => {
+                        param.glwe_dimension.0 * param.polynomial_size.0
+                    }
+                    PBSOrder::BootstrapKeyswitch => param.lwe_dimension.0,
+                };
+
+                message_modulus = param.message_modulus;
+                ciphertext_modulus = param.ciphertext_modulus;
+                carry_modulus = param.carry_modulus
+            }
+        };
+
+        let ct_len = self.ct.as_ref().len();
+
+        ct_len == expected_dim + 1
```

Comment:
I would prefer using the LweSize newtype

Reply:
or even use a lwe_ciphertext_size function (which probably does not exist) as we have size functions for other more complex ciphertext